enhancements/ingress/lb-subnet-selection-aws.md

```diff
@@ -0,0 +1,592 @@
+
+### Removing a deprecated feature
+
+N/A
+
+## Upgrade / Downgrade Strategy
+
+Upgrade strategy has been discussed in [Reason #2: To mitigate upgrade compatibility issues.](#reason-2-to-mitigate-upgrade-compatibility-issues).
+This design supports cluster admins who previously set the `service.beta.kubernetes.io/aws-load-balancer-subnets`
+annotation by not automatically impacting IngressController subnet configuration on cluster upgrade. Cluster
+admins will have to follow [Unmanaged Subnet Annotation Migration Workflow](#unmanaged-subnet-annotation-migration-workflow)
+in order to clear the `LoadBalancerProgressing` status condition message.
+
+Downgrades will also maintain compatibility. Downgrading to 4.(y-1) will preserve
+the current value of the `service.beta.kubernetes.io/aws-load-balancer-subnets` annotation.
+
+## Version Skew Strategy
+
+N/A
+
+## Operational Aspects of API Extensions
+
+N/A
+
+## Support Procedures
+
+### Multiple Subnets in Same AZ Support Procedure
+
+If a cluster admin has provided two subnets in the same Availability Zone (AZ), as
+discussed in [Invalid Subnet Annotation Values](#invalid-subnet-annotation-values),
+the IngressController with have a status with a relevant error:
+
+```yaml
+  - lastTransitionTime: "2024-04-02T22:07:59Z"
+    message: |-
+      The service-controller component is reporting SyncLoadBalancerFailed events like: Error syncing load balancer: failed to ensure load balancer: InvalidConfigurationRequest: ELB cannot be attached to multiple subnets in the same AZ.
+      The kube-controller-manager logs may contain more details.
+    reason: SyncLoadBalancerFailed
+    status: "False"
+    type: LoadBalancerReady
+```
+
+### Non-Existent Subnet Support Procedure
+
+If a cluster admin has provided a subnet that doesn't exist in the `Subnets` field,
+as discussed in [Invalid Subnet Annotation Values](#invalid-subnet-annotation-values),
+the IngressController with have a status with a relevant error:
+
+```yaml
+  - lastTransitionTime: "2024-04-02T22:07:59Z"
+    message: |-
+      The service-controller component is reporting SyncLoadBalancerFailed events like: Error syncing load balancer: failed to ensure load balancer: expected to find 1, but found 0 subnets
+      The kube-controller-manager logs may contain more details.
+    reason: SyncLoadBalancerFailed
+    status: "False"
+    type: LoadBalancerReady
+```
+
+## Alternatives
+
+### Automatically Recreating the Load Balancer
+
+One alternative to requiring cluster admins to recreate the LoadBalancer-type
+service, as described in [Effectuating Subnet Updates](#effectuating-subnet-updates),
+is to have the Ingress Operator automatically delete and recreate the service.
+
+While simpler in design, this approach could result in several minutes of unexpected
+disruption to ingress traffic. A cluster admin might not anticipate that updating
+the Subnets field could cause disruption, leading to an unwelcome surprise.
```

Comment:
If we make the field immutable and require a force update, that makes it clear to the admin no?

Reply:
Yep, it does it make it very clear that the Load Balancer needs to be deleted. But I'll refer to this [thread](https://github.com/openshift/enhancements/pull/1595/files#r1572193593) for the motivation.

Reply:
The link isn't working for me to see the other thread, was the thread resolved?

Reply:
Sorry github is odd sometimes. Hopefully this link works: https://github.com/openshift/enhancements/pull/1595#discussion_r1573445995

It's the thread with:

> So I actually wrote this EP originally to make the IngressController immutable. Then David [suggested](https://redhat-internal.slack.com/archives/CE4L0F143/p1712258802900769?thread_ts=1711471549.819729&cid=CE4L0F143) a mutable design. Miciah [summarized](https://redhat-internal.slack.com/archives/CE4L0F143/p1712351558628299?thread_ts=1711471549.819729&cid=CE4L0F143) the motivation to go mutable quite well.

> My understanding why mutability is better: